file_paths.md

```diff
@@ -204,6 +217,20 @@ user or group.
             run                               # :rundir                      same
                 agent.pid                     # :pidfile
 
+    C:\ProgramData\PuppetLabs\pxp-agent *
+        etc *
+            pxp-agent.cfg                     # pxp-agent configuration file
+            modules.d *                       # directory where pxp-agent modules are stored
+                puppet.cfg *                  # ../../var/modules/puppet configuration
+        var *
+            log
+                pxp-agent.log                 # enabled by default
+            modules *
+                puppet.bat *                  # executes Puppet and provides results of Puppet runs
```

Comment:
I'd expect pxp-agent to be able to execute `C:\Program Files\Puppet Labs\puppet\bin\puppet.bat` directly.

I think @MikaelSmith unit tests were that way, because on windows you can only execute a ruby script, e.g. `.\foo.rb` if there is a ruby application registered to handle files with that extension, which shows up in the `PATHEXT` environment variable. So he created a bat wrapper script to execute ruby with the script as an argument. Ruby itself does similar things with `gem.bat`, `irb.bat`, etc

Also the `%~dp0` magic is to get the directory that the .bat file is currently in, so that you can build a relative path and have it work if the files are relocated, e.g. user decides to install somewhere else.

Going back to the pxp-agent executing puppet question, I'm not sure you'll know where puppet is installed exactly, e.g. `C:\Program Files` vs `C:\Program Files (x86)` vs `D:\User\Specified\Path`. You could use a relative path based on the location of the pxp-agent binary (facter does something similar to locate ruby's dll/so to evaluate ruby-based facts, and that functionality is provided in leatherman) or assume `puppet` is in the PATH (and ensure the service is configured correctly).